go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -359,5 +389,14 @@ func allocatePodAnnotationWithRollback(
 		podAnnotation = tentative
 	}
 
+	if ipamClaim != nil && claimsReconciler != nil && needsAnnotationUpdate && err == nil {
```

Comment:
Suggestion: `if hasIPAMClaim && needsAnnotationUpdate && err == nil`

However, I think we potentially would need to reconcile the ipam claim even if `needsAnnotationUpdate` is false to cover up for cases where the ipam claim reference was added to the network selection element annotation after the IP are initally allocated. I know it is unexpected but I can't think of a reason why it would be better not to do it so...

Reply:
OK so after discussing this offline w/ you we've agreed on reconciling the IPAMClaim even when the pod was not updated. The normal flow is for the reconcile loop to be a no-op. 

But I realized I can't follow your suggestion; it would make it impossible to reconcile when the IPAMClaim exists but does not have IPs - which is the common scenario.